x-pack/plugins/osquery/public/live_queries/form/index.tsx

```diff
@@ -150,75 +195,18 @@ const LiveQueryFormComponent: React.FC<LiveQueryFormProps> = ({
     isLive,
   });
 
-  const { form } = useForm({
-    id: FORM_ID,
-    schema: liveQueryFormSchema,
-    onSubmit: async (formData, isValid) => {
-      if (isValid) {
-        try {
-          // @ts-expect-error update types
-          await mutateAsync(formData);
-          // eslint-disable-next-line no-empty
-        } catch (e) {}
-      }
-    },
-    options: {
-      stripEmptyFields: false,
-    },
-    serializer: ({
-      savedQueryId,
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      ecs_mapping,
-      packId,
-      ...formData
-    }) =>
-      pickBy(
-        {
-          ...formData,
-          pack_id: packId?.length ? packId[0] : undefined,
-          saved_query_id: savedQueryId,
-          ecs_mapping: convertECSMappingToObject(ecs_mapping),
-        },
-        (value) => !isEmpty(value)
-      ),
-  });
-
-  const { updateFieldValues, setFieldValue, submit, isSubmitting } = form;
-
   const actionId = useMemo(() => liveQueryDetails?.action_id, [liveQueryDetails?.action_id]);
   const agentIds = useMemo(() => liveQueryDetails?.agents, [liveQueryDetails?.agents]);
-  const [
-    { agentSelection, ecs_mapping: ecsMapping, query, savedQueryId, packId },
-    formDataSerializer,
-  ] = useFormData({
-    form,
-  });
 
-  /* recalculate the form data when ecs_mapping changes */
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  const serializedFormData = useMemo(() => formDataSerializer(), [ecsMapping, formDataSerializer]);
-
-  const agentSelected = useMemo(
-    () =>
-      agentSelection &&
-      !!(
-        agentSelection.allAgentsSelected ||
-        agentSelection.agents?.length ||
-        agentSelection.platformsSelected?.length ||
-        agentSelection.policiesSelected?.length
-      ),
-    [agentSelection]
-  );
-
-  const queryValueProvided = useMemo(() => !!query?.length, [query]);
+  const { ecs_mapping: ecsMapping, query, savedQueryId, packId } = watchedValues;
```

Comment:
can we rename the form field to `ecsMapping`, so we avoid renaming it everywhere?

Reply:
It takes a lot of effort. Let me first tackle all the rest, and then I will take another go at this one, would that be ok ? :)